lib/realtime/headway.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Realtime.Headway do
+  alias Gtfs.{Direction, Route, Stop}
+  alias Realtime.TimePeriod
+
+  @type headway_spacing ::
+          :very_bunched
+          | :bunched
+          | :ok
+          | :gapped
+          | :very_gapped
+
+  @type key_route_headways :: %{Route.id() => direction_origin_headways()}
+
+  @type direction_origin_headways :: %{Direction.id() => origin_headways()}
+
+  @type origin_headways :: %{Stop.id() => [time_period_headway()]}
+
+  @type time_period_headway :: %{
+          time_period_name: String.t(),
+          average_headway: String.t() | nil
+        }
+
+  @type seconds :: non_neg_integer()
+
+  @spec current_headway_spacing(seconds(), seconds()) :: headway_spacing()
+  def current_headway_spacing(headway_seconds, expected_headway_seconds)
+      when headway_seconds / expected_headway_seconds >= 2,
+      do: :very_gapped
+
+  def current_headway_spacing(headway_seconds, expected_headway_seconds)
+      when headway_seconds / expected_headway_seconds >= 1.5,
+      do: :gapped
+
+  def current_headway_spacing(headway_seconds, expected_headway_seconds)
+      when headway_seconds / expected_headway_seconds <= 0.33,
+      do: :very_bunched
+
+  def current_headway_spacing(headway_seconds, expected_headway_seconds)
```

Comment:
this is subjective, but i think it would be nice if this came before the previous function head so that the values go highest to lowest

Reply:
Would that would make it so that all bunched buses would get caught by the `<= 0.5` check and nothing would fall into the `<= 0.33` bucket?

Reply:
oh, right, of course it would, my mistake